### DISCUSSION TITLE
notify-function does not work properly

### DISCUSSION THREAD

**Discussion by jockele79** [svrooij/sonos2mqtt#203] [Q&A]
Hi  together,

I'm just playing around with the notify-command to run my doorbell by Sonos.
There are 2 main issues:

- in case of active playback the doorbell-mp3 is not played completely, it stops after few split seconds. When playing during inactive playback it works properly

- parameters "onlywhenplaying","timeout" and "delay" do not really have an influence, independently from changing these parameters the result is always the same. Is it possible to describe the expected action of these parameters more detailed?

I hope someone can e help,

thanks a lot

**Comment by svrooij** [comment]
What values are you setting for timeout and delay?
If you set delay to 800 and timeout to 15, things should work much better. But this only works if sonos2mqtt is receiving the events with updates.

If its not receiving the events, it does not know that it's playing or not. https://sonos2mqtt.svrooij.io/getting-started.html#events-explained

Option two, you enable the 'experimental' option and it will switch to "native" notification support. But still the onlyWhenPlaying option will only work if sonos2mqtt is receiving the events (see above)
https://sonos2mqtt.svrooij.io/getting-started.html#configuration

**Reply by svrooij** [reply]
@jockele79 can you try starting it with `experimental` set? This will start using native support for notifications. Which is way better then the solution we build into this library.

And if this solved your question, please mark the answer as the correct one.

**Reply by jockele79** [reply]
I just tested it, command was set with following parameters:

`{"command": "notifytwo", "input": {"trackUri": "x-file-cifs://192.168.178.5/multimedia/doorbell_11.mp3", "onlyWhenPlaying": "false", "timeout": 15, "volume": 100, "delayMS": 800}}`

Result: doorbell-clip is not played copletely,only a few split seconds. Results are exactly the same as described above

**Reply by svrooij** [reply]
I was not talking about notifytwo (that does not work correctly in my opinion)
I was talking about the notify command and the experimental features turned on with the environment variable SONOS2MQTT_EXPERIMENTAL=true set.

NotifyTwo is a secondary implementation build by another developer.

**Reply by jockele79** [reply]
sorry, I'm just a normal user, so I need your help again.
Activation of the experimental is clear, but what commands do I need to send???
I cannot find any documentation,

sorry and thanks

**Reply by svrooij** [reply]
Just the regular [notify](https://sonos2mqtt.svrooij.io/control/notifications.html#play-a-mp3) command, but the experimental implementation does not use timeout or delay, so this might be enough.

```json
{
  "command": "notify",
  "input": {
    "trackUri": "https://cdn.smartersoft-group.com/various/pull-bell-short.mp3",
    "onlyWhenPlaying": false,
    "volume": 40,
  }
}
```

Upon activating the experimental features, you should see something like this in the logs.
```txt
2024-02-07T08:05:35.259Z [Information] Starting sonos2mqtt v3.2.0-beta.14
2024-02-07T08:05:35.272Z [Information] LogLevel changed to information
2024-02-07T08:05:35.298Z [Information] Found ... sonos speakers
2024-02-07T08:05:35.298Z [Information] Experimental features activated, please provide feedback on notifications. https://github.com/svrooij/sonos2mqtt/discussions/191
```

**Comment by jockele79** [comment]
thanks for your feedback.
Parameters were set as timeout=20s and delay=800ms, I will check with timeout=15s. I will also check the experimental version and give feedback.

Concerning the events I think this works properly, with every change on the players I receive a message like the follwoing on the topic "sonos/RINCONXXX01400":

```json
{
  "uuid": "RINCON_XXXX01400",
  "model": "Sonos One",
  "name": "Esszimmer",
  "groupName": "Esszimmer",
  "coordinatorUuid": "RINCON_XXX01400",
  "currentTrack": {
    "Artist": "Eddie Money",
    "AlbumArtUri": "http://192.168.178.30:1400/getaa?s=1&u=x-sonosapi-stream%3atunein%253a9631%3fsid%3d303%26flags%3d8232%26sn%3d2",
    "Title": "Two tickets to paradise",
    "UpnpClass": "object.item",
    "ItemId": "-1",
    "ParentId": "-1",
    "TrackUri": "x-sonosapi-stream:tunein:9631?sid=303&flags=8232&sn=2",
    "ProtocolInfo": "sonos.com-http:*:application/octet-stream:*"
  },
  "enqueuedMetadata": {
    "Title": "ROCK ANTENNE",
    "UpnpClass": "object.item.audioItem.audioBroadcast",
    "ItemId": "-1",
    "ParentId": "-1"
  },
  "playmode": "NORMAL",
  "ts": 1710050196643,
  "transportState": "PLAYING",
  "volume": {
    "Master": 22,
    "LF": 100,
    "RF": 100
  },
  "mute": {
    "Master": false,
    "LF": false,
    "RF": false
  },
  "bass": 0,
  "treble": 0,
  "nextTrack": {
    "UpnpClass": "object.item.audioItem.musicTrack",
    "Duration": "0:00:00",
    "ItemId": "-1",
    "ParentId": "-1",
    "TrackUri": "https://d3rz60g1mtspwx.cloudfront.net/d417df26-cf55-4200000530339225/afc2402e-ec6f-3c82-8da5-fa0aa20ffd44.flac?Expires=1709977500&Signature=bKAr2W3fp4cwWyW62SPAFVfBMsMV7ZSe4uogQPtLc3CkFyXhrbaDBQ~FS009DjLi-OL0rIrp7j-8oyiMHSCUhrz1FEKPbymXVJhH7vFxsCExokEJ3tbZE7KiSl2d~RYawfOAplyu1X22~oTp-iAbg-HwEClNIBRckMXM8dq0WRs_&Key-Pair-Id=APKAJVZTZLZ7I5XDXGUQ"
  }
}
```
